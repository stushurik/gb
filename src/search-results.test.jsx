jest.mock('./search-results-loader.container', () => ({children}) => childrenu)

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SearchResults from './search-results';

describe('search results', () => {
  let props;

  beforeEach(() => {
    props = {
      users: []
    };
  });

  it('should show that no results were found', () => {
    const tree = renderer.create(<SearchResults {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show users if they were found', () => {
    props.users = [
      {
        id: '1',
        login: 'John Doe',
        avatar_url: 'http://example.com/fancy_kitten.png'
      },
      {
        id: '2',
        login: 'Darth Weider',
        avatar_url: 'http://example.com/death_start.png'
      },
      {
        id: '3',
        login: 'Bill Gates',
        avatar_url: 'http://example.com/windows.png'
      }
    ];
    const tree = renderer.create(<SearchResults {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('user selection', () => {
    let sut;

    beforeEach(() => {
      props.onUserSelect = jest.fn();
      props.users = [
        {
          id: '1',
          login: 'John Doe',
          avatar_url: 'http://example.com/fancy_kitten.png'
        },
        {
          id: '2',
          login: 'Darth Weider',
          avatar_url: 'http://example.com/death_start.png'
        },
        {
          id: '3',
          login: 'Bill Gates',
          avatar_url: 'http://example.com/windows.png'
        }
      ];

      sut = shallow(<SearchResults {...props} />);
    });

    it('should notify about user selection', () => {
      sut
        .find('[data-test="user"]')
        .first()
        .simulate('click');
      expect(props.onUserSelect).toHaveBeenCalledWith(props.users[0].login);
    });
  });
});
