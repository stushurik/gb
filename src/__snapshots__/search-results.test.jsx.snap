// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search results should show that no results were found 1`] = `
<span
  className="search-results"
>
  No results matching query were found
</span>
`;

exports[`search results should show users if they were found 1`] = `
<table
  className="search-results"
>
  <thead>
    <tr>
      <th>
        Avatar
      </th>
      <th>
        Name
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      data-test="user"
      onClick={[Function]}
    >
      <td>
        <img
          alt="avatar"
          src="http://example.com/fancy_kitten.png"
        />
      </td>
      <td>
        John Doe
      </td>
    </tr>
    <tr
      data-test="user"
      onClick={[Function]}
    >
      <td>
        <img
          alt="avatar"
          src="http://example.com/death_start.png"
        />
      </td>
      <td>
        Darth Weider
      </td>
    </tr>
    <tr
      data-test="user"
      onClick={[Function]}
    >
      <td>
        <img
          alt="avatar"
          src="http://example.com/windows.png"
        />
      </td>
      <td>
        Bill Gates
      </td>
    </tr>
  </tbody>
</table>
`;
